Gems/ROS2/Code/Source/Camera/CameraSensor.cpp

```diff
@@ -0,0 +1,240 @@
+            { AZ::RHI::Format::R8_UNORM, sizeof(uint8_t) },
+            { AZ::RHI::Format::R16_UNORM, sizeof(uint16_t) },
+            { AZ::RHI::Format::R32_FLOAT, sizeof(float) },
+        };
+
+    } // namespace Internal
+    CameraSensorDescription::CameraSensorDescription(const AZStd::string& cameraName, float verticalFov, int width, int height)
+        : m_verticalFieldOfViewDeg(verticalFov)
+        , m_verticalFieldOfViewRad(AZ::DegToRad(m_verticalFieldOfViewDeg))
+        , m_width(width)
+        , m_height(height)
+        , m_cameraName(cameraName)
+        , m_aspectRatio(static_cast<float>(width) / static_cast<float>(height))
+        , m_viewToClipMatrix(MakeViewToClipMatrix())
+        , m_cameraIntrinsics(MakeCameraIntrinsics())
+    {
+        validateParameters();
+    }
+
+    AZ::Matrix4x4 CameraSensorDescription::MakeViewToClipMatrix() const
+    {
+        const float nearDist = 0.1f, farDist = 100.0f;
+        AZ::Matrix4x4 localViewToClipMatrix;
+        AZ::MakePerspectiveFovMatrixRH(
+            localViewToClipMatrix, AZ::DegToRad(m_verticalFieldOfViewDeg), m_aspectRatio, nearDist, farDist, true);
+        return localViewToClipMatrix;
+    }
+
+    void CameraSensorDescription::validateParameters() const
+    {
+        AZ_Assert(
+            m_verticalFieldOfViewDeg > 0.0f && m_verticalFieldOfViewDeg < 180.0f,
+            "Vertical fov should be in range 0.0 < FoV < 180.0 degrees");
+        AZ_Assert(!m_cameraName.empty(), "Camera name cannot be empty");
+    }
+
+    AZStd::array<double, 9> CameraSensorDescription::MakeCameraIntrinsics() const
+    {
+        //  Intrinsic camera matrix of the camera image is being created here
+        //  It is based on other parameters available in the structure - they must be initialized before this function is called
+        //  Matrix is row-major and has the following form:
+        //  [fx  0 cx]
+        //  [ 0 fy cy]
+        //  [ 0  0  1]
+        // Projects 3D points in the camera coordinate frame to 2D pixel
+        // coordinates using the focal lengths (fx, fy) and principal point
+        // (cx, cy).
+        const auto w = static_cast<double>(m_width);
+        const auto h = static_cast<double>(m_height);
+        const double horizontalFoV = 2.0 * AZStd::atan(AZStd::tan(m_verticalFieldOfViewRad / 2.0) * m_aspectRatio);
+        const double focalLengthX = w / (2.0 * AZStd::tan(horizontalFoV / 2.0));
+        const double focalLengthY = h / (2.0 * AZStd::tan(m_verticalFieldOfViewRad / 2.0));
+        return { focalLengthX, 0.0, w / 2.0, 0.0, focalLengthY, h / 2.0, 0.0, 0.0, 1.0 };
+    }
+
+    CameraSensor::CameraSensor(const CameraSensorDescription& cameraSensorDescription)
+        : m_cameraSensorDescription(cameraSensorDescription)
+    {
+    }
+
+    void CameraSensor::setupPasses()
+    {
+        AZ_TracePrintf("CameraSensor", "Initializing pipeline for %s", m_cameraSensorDescription.m_cameraName.c_str());
```

Comment:
You may want to add a newline to the `AZ_TracePrintf` call, to make sure the next output statement starts on a newline.
```suggestion
        AZ_TracePrintf("CameraSensor", "Initializing pipeline for %s\n", m_cameraSensorDescription.m_cameraName.c_str());
```